src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -629,15 +635,15 @@ namespace nbl
 							const size_t accessorID = glTFprimitive.attributes.joints.value();
 
 							auto& glTFJointsXAccessor = glTF.accessors[accessorID];
-							handleAccessor(glTFJointsXAccessor, SAttributes::JOINTS_ATTRIBUTE_LAYOUT_ID); // TODO: ALWAYS UVEC4, PACK TO IT IF NECESSARY
+							handleAccessor(glTFJointsXAccessor, cpuMeshBuffer->getJointIDAttributeIx()); // TODO: ALWAYS UVEC4, PACK TO IT IF NECESSARY
 						}
 
 						if (glTFprimitive.attributes.weights.has_value())
 						{
 							const size_t accessorID = glTFprimitive.attributes.weights.value();
 
 							auto& glTFWeightsXAccessor = glTF.accessors[accessorID];
-							handleAccessor(glTFWeightsXAccessor, SAttributes::WEIGHTS_ATTRIBUTE_LAYOUT_ID); // TODO: ALWAYS UVEC4, PACK TO IT IF NECESSARY
+							handleAccessor(glTFWeightsXAccessor, cpuMeshBuffer->getJointWeightAttributeIx()); // TODO: ALWAYS UVEC4, PACK TO IT IF NECESSARY
```

Comment:
count how many joint influences you have per vertex, sort them by weight, keep maximum of 4

then use a vertex attribute as large as you need (max joints per vertex)

Reply:
each joint influence 1 up to 4 (weight either) is assigned to single vertex, and because there is an assignment I'm not sure if I can sort it, I would break the vertex-joint-weight hierarchy

Reply:
done